packages/indexer-agent/src/commands/common-options.ts

```diff
@@ -0,0 +1,147 @@
+      ) =>
+        arg.reduce(
+          (acc: string[], value: string) => [...acc, ...value.split(',')],
+          [],
+        ),
+      group: 'Indexer Infrastructure',
+    })
+    .option('indexer-management-port', {
+      description: 'Port to serve the indexer management API at',
+      type: 'number',
+      default: 8000,
+      required: false,
+      group: 'Indexer Infrastructure',
+    })
+    .option('metrics-port', {
+      description: 'Port to serve Prometheus metrics at',
+      type: 'number',
+      default: 7300,
+      required: false,
+      group: 'Indexer Infrastructure',
+    })
+    .option('syncing-port', {
+      description:
+        'Port to serve the network subgraph and other syncing data for indexer service at',
+      type: 'number',
+      default: 8002,
+      required: false,
+      group: 'Indexer Infrastructure',
+    })
+    .option('log-level', {
+      description: 'Log level',
+      type: 'string',
+      default: 'debug',
+      group: 'Indexer Infrastructure',
+    })
+    .option('offchain-subgraphs', {
+      description: 'Subgraphs to index that are not on chain (comma-separated)',
+      type: 'string',
+      array: true,
+      default: [],
+      coerce: arg =>
+        arg
+          .reduce(
+            (acc: string[], value: string) => [...acc, ...value.split(',')],
+            [],
+          )
+          .map((id: string) => id.trim())
+          .filter((id: string) => id.length > 0),
+    })
+    .option('postgres-host', {
+      description: 'Postgres host',
+      type: 'string',
+      required: true,
+      group: 'Postgres',
+    })
+    .option('postgres-port', {
+      description: 'Postgres port',
+      type: 'number',
+      default: 5432,
+      group: 'Postgres',
+    })
+    .option('postgres-username', {
+      description: 'Postgres username',
+      type: 'string',
+      required: false,
+      default: 'postgres',
+      group: 'Postgres',
+    })
+    .option('postgres-password', {
+      description: 'Postgres password',
+      type: 'string',
+      default: '',
+      required: false,
+      group: 'Postgres',
+    })
+    .option('postgres-database', {
+      description: 'Postgres database name',
+      type: 'string',
+      required: true,
+      group: 'Postgres',
+    })
+    .option('graph-node-query-endpoint', {
+      description: 'Graph Node endpoint for querying subgraphs',
+      type: 'string',
+      required: true,
+      group: 'Indexer Infrastructure',
+    })
+    .option('graph-node-status-endpoint', {
+      description: 'Graph Node endpoint for indexing statuses etc.',
+      type: 'string',
+      required: true,
+      group: 'Indexer Infrastructure',
+    })
+    .option('graph-node-admin-endpoint', {
+      description:
+        'Graph Node endpoint for applying and updating subgraph deployments',
+      type: 'string',
+      required: true,
+      group: 'Indexer Infrastructure',
+    })
+    .config({
+      key: 'config-file',
```

Comment:
To use network specificaiton files (for L2/multinetwork operation), you need to execute `indexer start-multiple` instead of just `start`. The new subcommand has a different argument parsing tree, in which you can specify the `--network-specifications-directory` option, that will read all network specification YAML files under a directory.

The `--config-file` option can be used in both subcommands, but the `indexerOptions.mnemonic` is specific to network specification files, thus, will only work if you use `start-multiple`. If you want to use a config file with `start`, then that option is named just `mnemonic`.